src/mjolnir/ingest_transit.cc

```diff
@@ -280,101 +369,127 @@ bool write_stop_pairs(Transit& tile,
     const std::string currFeedPath = feed_trip.feed;
     const auto& feed = tripFeeds(feed_trip);
 
-    gtfs::Trip currTrip = *(feed.get_trip(tile_tripId));
+    auto currTrip = feed.get_trip(tile_tripId);
 
     // already sorted by stop_sequence
-    auto tile_stopTimes = feed.get_stop_times_for_trip(tile_tripId);
+    const auto& tile_stopTimes = feed.get_stop_times_for_trip(tile_tripId);
```

Comment:
we need to check things like this. the api actually returns a copy here, not a reference. i think the compiler smooths this over for us but the reader of this code would make assumptions that wouldnt be true in this case.